edgelet/support-bundle/src/support_bundle old.rs

```diff
@@ -0,0 +1,684 @@
+        let mut inspect = ShellCommand::new("docker");
+
+        /***
+         * Note: just like inspect, this assumes using windows containers on a windows machine.
+         */
+        #[cfg(windows)]
+        inspect.args(&["-H", "npipe:////./pipe/iotedge_moby_engine"]);
+
+        inspect.args(&["network", "ls"]);
+        inspect.args(&["--format", "{{.Name}}"]);
+        let inspect = inspect.output();
+
+        let result = if let Ok(result) = inspect {
+            if result.status.success() {
+                String::from_utf8_lossy(&result.stdout).to_string()
+            } else {
+                println!(
+                    "Could not find network names: {}",
+                    String::from_utf8_lossy(&result.stderr)
+                );
+                "azure-iot-edge".to_owned()
+            }
+        } else {
+            println!("Could not find network names: {}", inspect.err().unwrap());
+            "azure-iot-edge".to_owned()
+        };
+
+        Ok((result.lines().map(String::from).collect(), self))
+    }
+
+    fn write_docker_network_to_file(mut self, network_name: &str) -> Result<Self, Error> {
+        self.print_verbose(&format!(
+            "Running docker network inspect for {}",
+            network_name
+        ));
+        let mut inspect = ShellCommand::new("docker");
+
+        /***
+         * Note: just like inspect, this assumes using windows containers on a windows machine.
+         */
+        #[cfg(windows)]
+        inspect.args(&["-H", "npipe:////./pipe/iotedge_moby_engine"]);
+
+        inspect.args(&["network", "inspect", &network_name, "-v"]);
+        let inspect = inspect.output();
+
+        let (file_name, output) = if let Ok(result) = inspect {
+            if result.status.success() {
+                (format!("network/{}.json", network_name), result.stdout)
+            } else {
+                (format!("network/{}_err.json", network_name), result.stderr)
+            }
+        } else {
+            let err_message = inspect.err().unwrap().to_string();
+            println!(
+                "Could not reach docker. Including error in bundle.\nError message: {}",
+                err_message
+            );
+            (
+                format!("network/{}_err_docker.txt", network_name),
+                err_message.as_bytes().to_vec(),
+            )
+        };
+
+        self.zip_writer
+            .start_file(file_name, self.file_options)
+            .map_err(|err| Error::from(err.context(ErrorKind::SupportBundle)))?;
+
+        self.zip_writer
+            .write_all(&output)
+            .map_err(|err| Error::from(err.context(ErrorKind::SupportBundle)))?;
+
+        self.print_verbose(&format!("Got docker network inspect for {}", network_name));
+        Ok(self)
+    }
+
+    fn print_verbose(&self, message: &str) {
+        if self.verbose {
+            println!("{}", message);
+        }
+    }
```

Comment:
I noticed elsewhere we have a similar function; could we just have one version/style and use that? my 2 cents is to use standalone functions, that way if you ever want `print_warn` or similar you can add it easier :)